File: Labs/Lab-04-20231013/app/src/main/java/sap/escooters/data_source_layer/DataSourceLayer.java
package sap.escooters.data_source_layer;

import sap.layers.Layer;
import io.vertx.core.json.JsonObject;

public interface DataSourceLayer extends Layer {

	void saveUser(JsonObject user) throws DataSourceException;
	void saveEScooter(JsonObject scooter) throws DataSourceException;
	void saveRide(JsonObject ride) throws DataSourceException;
}

File: Labs/Lab-05-20231016/app/src/main/kotlin/assigment03/escooters/data_source_layer/DataSourceLayer.kt
package assigment03.escooters.data_source_layer

import assigment03.layers.Layer
import io.vertx.core.json.JsonObject

interface DataSourceLayer : Layer {
    @Throws(DataSourceException::class)
    fun saveUser(user: JsonObject)

    @Throws(DataSourceException::class)
    fun saveEScooter(escooter: JsonObject)

    @Throws(DataSourceException::class)
    fun saveRide(ride: JsonObject)
}